server/routes/v1/version.ts

```diff
@@ -138,3 +140,56 @@ export const updateLastViewed = [
     return res.json({ version: id, role })
   },
 ]
+
+export const deleteVersion = [
+  ensureUserRole('user'),
+  async (req: Request, res: Response) => {
+    const { id } = req.params
+    const { user } = req
+
+    const version = await findVersionById(user, id)
+
+    if (!version) {
+      throw NotFound({ code: 'version_not_found', id }, 'Unable to find version')
+    }
+
+    if (req.user.id !== version.metadata.contacts.uploader) {
+      throw Forbidden({ code: 'user_unauthorised' }, 'User is not authorised to do this operation.')
+    }
+
+    const versionRequests = await RequestModel.find({ version: version._id })
+    if (versionRequests.length > 0) {
+      await Promise.all(versionRequests.map((versionRequest) => versionRequest.delete()))
+    }
+
+    const deployments = await DeploymentModel.find({ versions: { $in: [id] } })
+    if (deployments.length > 0) {
```

Comment:
There's no reason for this, you can just `forEach` over an empty array to do nothing.